include/lsst/afw/typehandling/python.h

```diff
@@ -84,10 +88,121 @@ class StorableHelper : public Base {
     }
+
+    bool isPersistable() const noexcept override {
+        PYBIND11_OVERLOAD(
+            bool, Base, isPersistable
+        );
+    }
+
+    std::string getPersistenceName() const override {
+        PYBIND11_OVERLOAD_NAME(
+            std::string, Base, "_getPersistenceName", getPersistenceName
+        );
+    }
+
+    std::string getPythonModule() const override {
+        PYBIND11_OVERLOAD_NAME(
+            std::string, Base, "_getPythonModule", getPythonModule
+        );
+    }
+
+    void write(table::io::OutputArchiveHandle& handle) const override;
 };
 
 std::string declareGenericMapRestrictions(std::string const& className, std::string const& keyName);
 
+/**
+ * StorableHelper persistence.
+ *
+ * We cannot directly override `StorableHelper::write` or `StorableHelperFactory::read` in python using
+ * PYBIND11_OVERLOAD* macros as the required argument c++ types (OutputArchiveHandle, InputArchive,
+ * CatalogVector) are not bound to python types.  Instead, we allow python subclasses of Storable to define
+ * `_write` and `_read` methods that return/accept a string serialization of the object.  The `_write` method
+ * should take no arguments (besides self) and return a string.  The `_read` method should be a static method
+ * that takes a string and returns an instance of the python class.  The python pickle module may be useful
+ * for handling the string serialization / deserialization.
+ *
+ * The `StorableHelper::write` and `StorableHelperFactory::read` c++ methods will directly call the python
+ * methods described above when required.
+ */
+
+namespace {
+
+class StorableHelperPersistenceHelper {
+public:
+    table::Schema schema;
+    table::Key<table::Array<std::uint8_t>> bytes;
+
+    static StorableHelperPersistenceHelper const &get() {
+        static StorableHelperPersistenceHelper instance;
+        return instance;
+    }
+
+    // No copying
+    StorableHelperPersistenceHelper(StorableHelperPersistenceHelper const &) = delete;
+    StorableHelperPersistenceHelper &operator=(StorableHelperPersistenceHelper const &) = delete;
+
+    // No moving
+    StorableHelperPersistenceHelper(StorableHelperPersistenceHelper &&) = delete;
+    StorableHelperPersistenceHelper &operator=(StorableHelperPersistenceHelper &&) = delete;
+
+private:
+    StorableHelperPersistenceHelper() :
+        schema(),
+        bytes(schema.addField<table::Array<std::uint8_t>>(
+            "bytes", "an opaque bytestring representation of a Storable", ""
+        ))
+    {}
+};
+
+
+class StorableHelperFactory : public table::io::PersistableFactory {
+public:
+    StorableHelperFactory(std::string const &module, std::string const &name) :
+        table::io::PersistableFactory(name),
+        _module(module),
+        _name(name)
+    {}
+
+    std::shared_ptr<table::io::Persistable> read(
+        InputArchive const &archive,
+        CatalogVector const &catalogs
+    ) const override {
+        // Do I need to acquire the GIL here?
```

Comment:
Yes, you do.  Glad you thought about this, because I wouldn't have if you hadn't brought it up.